static/js/settings_org.js

```diff
@@ -955,94 +955,75 @@ exports.build_page = function () {
+        delete_button.show();
     }
 
-    function realm_icon_upload_start() {
-        $('#upload_icon_spinner').css({visibility: "visible"});
-        $('#realm_icon_upload').hide();
-        $('#realm_icon_delete_button').hide();
+    function realm_icon_logo_upload_start(spinner, upload_text, delete_button) {
+        spinner.css({visibility: "visible"});
+        upload_text.hide();
+        delete_button.hide();
     }
 
-    function upload_realm_icon(file_input) {
-        const form_data = new FormData();
-
-        form_data.append('csrfmiddlewaretoken', csrf_token);
-        for (const [i, file] of Array.prototype.entries.call(file_input[0].files)) {
-            form_data.append('file-' + i, file);
-        }
-
-        const error_field = $("#realm_icon_file_input_error");
-        error_field.hide();
-        realm_icon_upload_start();
-
-        channel.post({
-            url: '/json/realm/icon',
-            data: form_data,
-            cache: false,
-            processData: false,
-            contentType: false,
-            success: function () {
-                realm_icon_upload_complete();
-            },
-            error: function (xhr) {
-                realm_icon_upload_complete();
-                ui_report.error("", xhr, error_field);
-            },
-        });
-
-    }
-    realm_icon.build_realm_icon_widget(upload_realm_icon);
-
-    function upload_realm_logo(file_input, night) {
+    function upload_realm_logo_or_icon(file_input, night, icon) {
         const form_data = new FormData();
         let spinner;
         let error_field;
-        let button_text;
+        let upload_text;
+        let delete_button;
+        let url;
 
         form_data.append('csrfmiddlewaretoken', csrf_token);
         for (const [i, file] of Array.prototype.entries.call(file_input[0].files)) {
             form_data.append('file-' + i, file);
         }
-        if (night) {
-            error_field = $("#night-logo-section .realm-logo-file-input-error");
-            spinner = $("#night-logo-section .upload-logo-spinner");
-            button_text = $("#night-logo-section .upload-logo-button-text");
+        if (icon) {
+            url = '/json/realm/icon';
+            spinner = $('#icon-spinner-background').expectOne();
+            upload_text = $('#realm_icon_upload').expectOne();
+            delete_button = $('#realm_icon_delete_button').expectOne();
+            error_field = $("#realm_icon_file_input_error");
         } else {
-            error_field = $("#day-logo-section .realm-logo-file-input-error");
-            spinner = $("#day-logo-section .upload-logo-spinner");
-            button_text = $("#day-logo-section .upload-logo-button-text");
+            if (night) {
+                error_field = $("#night-logo-section .realm-logo-file-input-error");
+                spinner = $("#night-logo-section #logo-spinner-background");
+                upload_text = $('#night-logo-section #realm_logo_upload_text');
```

Comment:
We'll want these `#realm_logo_upload_text` elements to be classes, not IDs, since IDs should be unique in the DOM, and there's obviously 2 of them inside different sections.

Maybe just to make clear the ultimate goal here, since maybe there's a better path to get there: We ultimately will want an `image_upload_widget` similar to the `dropdown_list_widget`, that can use the same HTML template for all 4 widgets across the two settings UIs, with conditionals and clever selectors (E.g. this could be `$(section + " .image_upload_widget_text")` and be identical code between all the widgets.

@aero31aero can you help @Jagansivam28 out with getting this work into a mergable state?

I'm really happy with the UI for this, but I'd like us to end up with an implementation we really like here as well.